### PR TITLE
Feature/Add Template Message feature

### PR DESCRIPTION
**Pull request by pauloabreu**

### CODE REVIEW

File: marketplace/wpp_templates/models.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    )
+
+    TEMPLATE_TYPES_CHOICES = (
+        ("MEDIA", "WhatsApp.data.templates.type.media"),
+        ("INTERACTIVE", "WhatsApp.data.templates.type.interactive"),
+        ("TEXT", "WhatsApp.data.templates.type.text"),
+    )
+
+    uuid = models.UUIDField(default=uuid.uuid4, editable=False, unique=True)
+
```
Comment (elitonzky) [inline marketplace/wpp_templates/models.py:37]: Excess blank lines at: 37, 39, 41, 43, 45, 47. 
Extra blank lines can be used (sparingly) to separate groups of related functions.
More details in: https://peps.python.org/pep-0008/#should-a-line-break-before-or-after-a-binary-operator

File: marketplace/wpp_templates/tests/test_models.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            template=self.template_message, status="APPROVED", language="pt_br", variable_count=1
+        )
+        super().setUp()
+
+    def test_create_template_button_url(self):
+        TemplateButton.objects.create(
+            translation=self.template_translation, button_type="PHONE_NUMBER", country_code=55
+        )
+
+    """
```
Comment (elitonzky) [inline marketplace/wpp_templates/tests/test_models.py:42]: Is this commented part necessary? if so please put it as #TODO.

File: marketplace/wpp_templates/tests/test_models.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            template_type="TEXT",
+            created_by_id=User.objects.get_admin_user().id,
+        )
+        self.template_translation = TemplateTranslation.objects.create(
+            template=self.template_message, status="APPROVED", language="pt_br", variable_count=1
+        )
+        super().setUp()
+
+    """
+    def test_create_template_header_text_required_fail(self):
```
Comment (elitonzky) [inline marketplace/wpp_templates/tests/test_models.py:82]: Is this commented part necessary? if so please put it as #TODO.

File: marketplace/wpp_templates/tests/test_views.py [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+        super().setUp()
+
+    @property
+    def view(self):
+        return self.view_class.as_view({"get": "list"})
+
+    def test_list_whatsapp_templates(self):
+        response = self.request.get(self.url, app_uuid=str(self.app.uuid))
+
+        # print(response.json)
```
Comment (elitonzky) [inline marketplace/wpp_templates/tests/test_views.py:64]: forgot to remove?

File: marketplace/wpp_templates/tests/test_views.py [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+            template=self.template_message,
+            status="APPROVED",
+            language="pt_br",
+            variable_count=1,
+            body="Teste",
+        )
+
+        TemplateHeader.objects.create(translation=self.template_translation, header_type="TEXT", text="teste")
+
+        print(self.template_translation.headers)
```
Comment (elitonzky) [inline marketplace/wpp_templates/tests/test_views.py:53]: forgot to remove the print?

File: marketplace/wpp_templates/tests/test_views.py [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+        return self.view_class.as_view(APIBaseTestCase.ACTION_RETRIEVE)
+
+    @property
+    def query_parameter(self):
+        return f"?uuid={self.template_message.uuid}"
+
+    def test_retrieve_whatsapp_template(self):
+        response = self.request.get(self.url, app_uuid=self.app.uuid, uuid=self.template_message.uuid)
+
+        print(response.json)
```
Comment (elitonzky) [inline marketplace/wpp_templates/tests/test_views.py:188]: forgot to remove the print?

File: marketplace/wpp_templates/tests/test_views.py [hunk 7]
```diff
@@ -0,0 +1,257 @@
...
...XQgbm90IGFzIGJvcmluZyBhcyB3YXRjaGluZyApIFRqDQpFVA0KQlQNCi9GMSAwMDEwIFRmDQo2OS4yNTAwIDY1Mi43NTIwIFRkDQooIHBhaW50IGRyeS4gQW5kIG1vcmUgdGV4dC4gQW5kIG1vcmUgdGV4dC4gQW5kIG1vcmUgdGV4dC4gQW5kIG1vcmUgdGV4dC4gKSBUag0KRVQNCkJUDQovRjEgMDAxMCBUZg0KNjkuMjUwMCA2NDAuODAwMCBUZA0KKCBCb3JpbmcuICBNb3JlLCBhIGxpdHRsZSBtb3JlIHRleHQuIFRoZSBlbmQsIGFuZCBqdXN0IGFzIHdlbGwuICkgVGoNCkVUDQplbmRzdHJlYW0NCmVuZG9iag0KDQo4IDAgb2JqDQpbL1BERiAvVGV4dF0NCmVuZG9iag0KDQo5IDAgb2JqDQo8PA0KL1R5cGUgL0ZvbnQNCi9TdWJ0eXBlIC9UeXBlMQ0KL05hbWUgL0YxDQovQmFzZUZvbnQgL0hlbHZldGljYQ0KL0VuY29kaW5nIC9XaW5BbnNpRW5jb2RpbmcNCj4+DQplbmRvYmoNCg0KMTAgMCBvYmoNCjw8DQovQ3JlYXRvciAoUmF2ZSBcKGh0dHA6Ly93d3cubmV2cm9uYS5jb20vcmF2ZVwpKQ0KL1Byb2R1Y2VyIChOZXZyb25hIERlc2lnbnMpDQovQ3JlYXRpb25EYXRlIChEOjIwMDYwMzAxMDcyODI2KQ0KPj4NCmVuZG9iag0KDQp4cmVmDQowIDExDQowMDAwMDAwMDAwIDY1NTM1IGYNCjAwMDAwMDAwMTkgMDAwMDAgbg0KMDAwMDAwMDA5MyAwMDAwMCBuDQowMDAwMDAwMTQ3IDAwMDAwIG4NCjAwMDAwMDAyMjIgMDAwMDAgbg0KMDAwMDAwMDM5MCAwMDAwMCBuDQowMDAwMDAxNTIyIDAwMDAwIG4NCjAwMDAwMDE2OTAgMDAwMDAgbg0KMDAwMDAwMjQyMyAwMDAwMCBuDQowMDAwMDAyNDU2IDAwMDAwIG4NCjAwMDAwMDI1NzQgMDAwMDAgbg0KDQp0cmFpbGVyDQo8PA0KL1NpemUgMTENCi9Sb290IDEgMCBSDQovSW5mbyAxMCAwIFINCj4+DQoNCnN0YXJ0eHJlZg0KMjcxNA0KJSVFT0YNCg=="
```
Comment (elitonzky) [inline marketplace/wpp_templates/tests/test_views.py:234]: For such a long string it is advisable to store it in a separate file.
The variable "f", must be renamed according to its purpose.

**Review by elitonzky — CHANGES_REQUESTED** [review]
Pending changes

File: marketplace/wpp_templates/models.py (hunk 1 above)
Comment (pauloabreu) [reply marketplace/wpp_templates/models.py]: Done!

File: marketplace/wpp_templates/tests/test_models.py (hunk 3 above)
Comment (pauloabreu) [reply marketplace/wpp_templates/tests/test_models.py]: Done!

File: marketplace/wpp_templates/tests/test_views.py (hunk 5 above)
Comment (pauloabreu) [reply marketplace/wpp_templates/tests/test_views.py]: Done!

File: marketplace/wpp_templates/tests/test_views.py (hunk 7 above)
Comment (pauloabreu) [reply marketplace/wpp_templates/tests/test_views.py]: Done!

**Pull request closed without merging**